projectile.el

```diff
@@ -5118,7 +5118,13 @@ Otherwise behave as if called interactively.
     (add-hook 'projectile-find-dir-hook #'projectile-track-known-projects-find-file-hook t)
     (add-hook 'dired-before-readin-hook #'projectile-track-known-projects-find-file-hook t t)
     (advice-add 'compilation-find-file :around #'compilation-find-file-projectile-find-compilation-buffer)
-    (advice-add 'delete-file :before #'delete-file-projectile-remove-from-cache))
+    (advice-add 'delete-file :before #'delete-file-projectile-remove-from-cache)
+    (if (bound-and-true-p savehist-loaded)
```

Comment:
Oh second, thought - this is not ideal as we'll keep doing the same thing every time the mode starts. Probably it's better to have this at the top-level as initially proposed, just add some comments there explaining the need for this, as it's unlikely that even I'll remember this several months down the road.

Reply:
As I see, even we keep doing the same thing when mode starts but seems it doesn't affect anything. Anyway, up to you. Do you want to revert this commit?

Reply:
HI @bbatsov, any chance?

Reply:
Yeah, I think it's better to revert it and document better the previous solution.

Reply:
I updated. Hope it is good.